build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/transport/TransportVersionResourcesPlugin.java

```diff
@@ -49,6 +49,10 @@ public void apply(Project project) {
                 Directory transportResources = project.getLayout().getProjectDirectory().dir("src/main/resources/" + resourceRoot);
                 spec.getParameters().getTransportResourcesDirectory().set(transportResources);
                 spec.getParameters().getRootDirectory().set(project.getLayout().getSettingsDirectory().getAsFile());
+                Object upstreamRef = project.findProperty("org.elasticsearch.transport.upstreamRef");
```

Comment:
@breskeby is using `project.getProviders().gradleProperty()` preferable wrt to configuration caching or any of that?

Reply:
yeah I think we should use `project.getProviders().gradleProperty()` here indeed.